### ISSUE TITLE
New Prometheus alert firing

### ISSUE THREAD

**Issue by budimanjojo** [rook/rook#11785]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:** Updating to helm release `v1.11.0` with monitoring enabled produces a new prometheus alert `PrometheusJobMissing`. This alert is new: https://github.com/rook/rook/blame/08ec3a6e52b2fb3b4e89a9b4283de620a76c22c7/deploy/charts/rook-ceph-cluster/prometheus/localrules.yaml#L586-L597

**Expected behavior:** No alert

**Comment by parth-gr** [comment]
Looks like we need to update promotheus.yaml https://github.com/rook/rook/blob/08ec3a6e52b2fb3b4e89a9b4283de620a76c22c7/deploy/examples/monitoring/prometheus.yaml with https://docs.ceph.com/en/quincy/mgr/prometheus/#prometheus-yml

@[ahgraber](https://github.com/rook/rook/commits?author=ahgraber) for more info

**Comment by travisn** [comment]
For reference, this appears related to #11721

**Comment by ammmze** [comment]
I remember we had this before and then we removed some of those alerts ([here](https://github.com/rook/rook/pull/10109)). We must have pulled a new copy of the alerts from upstream ceph again?

**Comment by travisn** [comment]
@ammmze Thanks for the reminder, we did miss removing those during the alerts update.

**Comment by reefland** [comment]
Upgraded to v1.11.1 - still getting this Prometheus job missing alert.
```
$ rook version
rook: v1.11.1
go: go1.19.6
```
![image](https://user-images.githubusercontent.com/16230914/224374129-241c82ee-cbda-44be-aba6-ddf86b895f10.png)

Any manual cleanup needed?

Operator `values.yaml` has:
```
          monitoring:
            # requires Prometheus to be pre-installed
            # enabling will also create RBAC rules to allow Operator to create ServiceMonitors
            enabled: true
```

Cluster `values.yaml` has:
```
          # monitoring requires Prometheus to be pre-installed
          monitoring:
            # enabling will also create RBAC rules to allow Operator to create ServiceMonitors
            enabled: true
            # whether to create the prometheus rules
            createPrometheusRules: true
            # the namespace in which to create the prometheus rules, if different from the rook cluster namespace
            # If you have multiple rook-ceph clusters in the same k8s cluster, choose the same namespace (ideally, namespace with prometheus
            # deployed) to set rulesNamespace for all the clusters. Otherwise, you will get duplicate alerts with multiple alert definitions.
            rulesNamespaceOverride:
            # Monitoring settings for external clusters:
            # externalMgrEndpoints: <list of endpoints>
            # externalMgrPrometheusPort: <port>
            # allow adding custom labels and annotations to the prometheus rule
            prometheusRule:
              labels: {}
              annotations: {}
```

**Comment by travisn** [comment]
Did you update the cluster chart as well? If you only updated the rook-ceph chart to v1.11.1, but not the rook-ceph-cluster chart, it could explain that the alert is still there. Since #11843 was included in v1.11.1, I don't see why it would still be there otherwise.

**Comment by reefland** [comment]
Yeah, I did update both charts (RenovateBot opened PR on both charts).

I forced sync'd the Prometheus Rule object and it cleared up (ArgoCD).  I had enabled ServerSide apply on both charts to resolve a CRD issue in 1.10.0.  Likely didn't need to do that for the cluster chart. I don't know that was the cause, but best guess at the moment.  I'll keep an eye on things with next version. 

Thanks again for the prompt response.
